storage/src/test/java/tech/pegasys/teku/storage/server/DepositStorageTest.java

```diff
@@ -0,0 +1,155 @@
+    DepositsFromBlockEvent blockEvent2 = dataStructureUtil.randomDepositsFromBlockEvent(102L, 10);
+    MinGenesisTimeBlockEvent genesis = dataStructureUtil.randomMinGenesisTimeBlockEvent(101L);
+
+    database.addDepositsFromBlockEvent(blockEvent);
+    database.addDepositsFromBlockEvent(blockEvent2);
+    database.addMinGenesisTimeBlock(genesis);
+
+    SafeFuture<ReplayDepositsResult> future = depositStorage.replayDepositEvents();
+    assertThat(future.isDone()).isTrue();
+    assertThat(eventsChannel.getOrderedList()).containsExactly(blockEvent, genesis, blockEvent2);
+    assertThat(eventsChannel.getGenesis()).isEqualToComparingFieldByField(genesis);
+    assertThat(future.get().getBlockNumber().get()).isEqualTo(blockEvent2.getBlockNumber());
+    assertThat(future.get().isPastGenesisBlock()).isTrue();
+    depositStorage.stop();
+  }
+
+  @Test
+  public void shouldReplayMultipleDeposits() throws ExecutionException, InterruptedException {
+    DepositsFromBlockEvent blockEvent = dataStructureUtil.randomDepositsFromBlockEvent(100L, 10);
+    DepositsFromBlockEvent blockEvent2 = dataStructureUtil.randomDepositsFromBlockEvent(101L, 10);
+    database.addDepositsFromBlockEvent(blockEvent);
+    database.addDepositsFromBlockEvent(blockEvent2);
+
+    SafeFuture<ReplayDepositsResult> future = depositStorage.replayDepositEvents();
+    assertThat(future.isDone()).isTrue();
+    assertThat(eventsChannel.getOrderedList()).containsExactly(blockEvent, blockEvent2);
+    assertThat(eventsChannel.getGenesis()).isNull();
+    assertThat(future.get().getBlockNumber().get()).isEqualTo(blockEvent2.getBlockNumber());
+    assertThat(future.get().isPastGenesisBlock()).isFalse();
+    depositStorage.stop();
+  }
+
+  @Test
+  public void shouldSendBlockThenGenesisWhenBlockNumberIsTheSame() {
+    DepositsFromBlockEvent blockEvent = dataStructureUtil.randomDepositsFromBlockEvent(100L, 10);
+    MinGenesisTimeBlockEvent genesis = dataStructureUtil.randomMinGenesisTimeBlockEvent(100L);
+    database.addDepositsFromBlockEvent(blockEvent);
+    database.addMinGenesisTimeBlock(genesis);
+
+    SafeFuture<ReplayDepositsResult> future = depositStorage.replayDepositEvents();
+    assertThat(future.isDone()).isTrue();
+    assertThat(eventsChannel.getOrderedList()).containsExactly(blockEvent, genesis);
+    assertThat(eventsChannel.getGenesis()).isEqualToComparingFieldByField(genesis);
+    depositStorage.stop();
+  }
+
+  @Test
+  public void shouldJustSendGenesis() {
+    MinGenesisTimeBlockEvent genesis = dataStructureUtil.randomMinGenesisTimeBlockEvent(100L);
+    database.addMinGenesisTimeBlock(genesis);
+
+    SafeFuture<ReplayDepositsResult> future = depositStorage.replayDepositEvents();
+    assertThat(future.isDone()).isTrue();
+    assertThat(eventsChannel.getOrderedList()).containsExactly(genesis);
+    assertThat(eventsChannel.getGenesis()).isEqualToComparingFieldByField(genesis);
```

Comment:
Worth checking that the `ReplayDepositsResult` has the genesis block as the block to start from and true for `isPastGenesisBlock`

Reply:
picking up the last block number from min genesis event I wasn't aware of, so i'll need to fix the response